Add test for InstanceUploaderTask.doInBackground()

alxndrsn opened the pull request.

alxndrsn committed:
Add test for InstanceUploaderTask.doInBackground()

Test cases recorded against ODK Aggregate.

lognaturel commented:
@alxndrsn it looks like I ended up running the wrong version locally and didn't notice that calls to `firstRequestFor` in `WebUtilsTest` need to be replace. That's easy enough but in addition to that, the instance upload test fails because `</submission?deviceID=android_id%3A5d65a4705fa192c3>` and `</submission\?deviceID=imei%3A\d+>` don't match.

alxndrsn commented:
> it looks like I ended up running the wrong version locally and didn't notice that calls to firstRequestFor in WebUtilsTest need to be replace.

Sorry, looks like my `master` was out of date yesterday.  Fixed in https://github.com/opendatakit/collect/pull/929

> the instance upload test fails because </submission?deviceID=android_id%3A5d65a4705fa192c3> and </submission\?deviceID=imei%3A\d+> don't match.

Ah yes, well spotted!  I'm guessing you don't have a SIM card.  This check should be much more lenient.  Fixed in https://github.com/opendatakit/collect/pull/930

lognaturel commented:
No big deal, apparently we were all having some branching problems!

You're right, no SIM card was it. Interesting!
